.github/workflows/lint.yml

```diff
@@ -20,3 +20,7 @@ jobs:
         with:
           python-version: "3.x"
       - uses: tox-dev/action-pre-commit-uv@v1
+      - uses: astral-sh/ruff-action@v3
+        with:
+          args: check
+          src: "."
```

Comment:
We have pre-commit, let's add it there instead. It'll also be run on CI.

```suggestion
```

And add this to the top of `.pre-commit-config.yaml`:

```yml
repos:
  - repo: https://github.com/astral-sh/ruff-pre-commit
    rev: v0.12.3
    hooks:
      - id: ruff-check
        args: [--exit-non-zero-on-fix]
```